megamek/src/megamek/client/ui/swing/BoardEditor.java

```diff
@@ -113,6 +113,11 @@ public String getTerrainTooltip() {
         public int compareTo(TerrainHelper o) {
             return toString().compareTo(o.toString());
         }
+        
+        @Override
+        public boolean equals(Object o) {
+            return o instanceof TerrainHelper && this.getTerrainType() == ((TerrainHelper) o).getTerrainType();
```

Comment:
Should probably add a hashCode implementation that returns the hash of terrainType.

Reply:
What would be the point of adding a hashCode? It doesn't appear to be used at all in the JList implementation.

Reply:
Consistency/OCD? [Rummaging through OpenJDK it doesn't appear they do to your point](https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/jdk/src/share/classes/javax/swing/JList.java).

Reply:
Generally not a big fan of adding dead code, especially for a one-off internal class.